frame/staking/src/lib.rs

```diff
@@ -1427,7 +1478,13 @@ decl_module! {
 				ledger = ledger.consolidate_unlocked(current_era)
 			}
 
-			let post_info_weight = if ledger.unlocking.is_empty() && ledger.active < T::Currency::minimum_balance() {
+			let is_validator = Validators::<T>::contains_key(&stash);
+
+			// The minimum bond allowed for a stash before we clean it up.
+			// Requirements to be a validator are likely higher than normal.
+			let min_bond = if is_validator { T::MinValidatorBond::get() } else { T::MinBond::get() };
```

Comment:
well, maybe. Most validators have almost nothing at stake and move their bond to a self-nomination. 

But there are discussions around this as well https://github.com/w3f/research-security-issues/issues/58